### PR TITLE
feat(facet ordering): Dynamic facet ordering support

### PR DESCRIPTION
**Pull request by VladislavFitz**
**Summary**

Introduces the models for  dynamic facet ordering support

### CODE REVIEW

File: Sources/AlgoliaSearchClient/Command/AlgoliaCommand.swift
Comment (Haroenv) [inline Sources/AlgoliaSearchClient/Command/AlgoliaCommand.swift:23]: are all these stylistic changes expected?

File: Sources/AlgoliaSearchClient/Models/Rule/RenderingContent.swift
Comment (Haroenv) [inline Sources/AlgoliaSearchClient/Models/Rule/RenderingContent.swift:17]: let's not add this yet

File: Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/RenderingContent/FacetOrderContainer.swift
Comment (Haroenv) [inline Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/RenderingContent/FacetOrderContainer.swift:26]: is this an object or an array?

File: Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/RenderingContent/Redirect.swift
Comment (Haroenv) [inline Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/RenderingContent/Redirect.swift:11]: this should also be done in a separate PR

**Review by Haroenv — COMMENTED** [review]
schema looks correct

File: Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/RenderingContent/FacetOrderContainer.swift
Comment (VladislavFitz) [reply Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/RenderingContent/FacetOrderContainer.swift:26]: It's a map `Attribute -> FacetValuesOrder`

File: Sources/AlgoliaSearchClient/Models/Rule/RenderingContent.swift
Comment (VladislavFitz) [reply Sources/AlgoliaSearchClient/Models/Rule/RenderingContent.swift]: Removed 👌

**Review by Haroenv — APPROVED** [review]
my approval is only for the shape of the result / rule, I wouldn't know enough swift for the other areas of the code

**Comment by VladislavFitz** [comment]
> my approval is only for the shape of the result / rule, I wouldn't know enough swift for the other areas of the code

Sure, waiting for @Aallam approval.

File: Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/RenderingContent/FacetOrderContainer.swift
Comment (Aallam) [inline Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/RenderingContent/FacetOrderContainer.swift:10]: Is this still used? is`RenderingContent` enough?

File: Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/Facet/AttributedFacets.swift
Comment (Aallam) [inline Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/Facet/AttributedFacets.swift:10]: Should we moved this class to InstantSearch since the client doesn't use it?

Comment (Haroenv) [reply Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/Facet/AttributedFacets.swift:10]: wouldn't people not using InstantSearch still need to use it?

Comment (Aallam) [reply Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/Facet/AttributedFacets.swift:10]: Probably yes, however, to maintain the separation of concerns, I don't think the API client should be providing it.
You think we should do anyway?

File: Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/RenderingContent/FacetOrderContainer.swift
Comment (VladislavFitz) [reply Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/RenderingContent/FacetOrderContainer.swift]: Good catch, it's obsolete. Removed it.

**Review by Aallam — APPROVED** [review]

File: Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/Facet/AttributedFacets.swift
Comment (VladislavFitz) [reply Sources/AlgoliaSearchClient/Models/Search/Response/SearchResponse/Auxiliary/Facet/AttributedFacets.swift]: Good point, I moved it to the [InstantSearch PR](https://github.com/algolia/instantsearch-ios/pull/168). 
This structure is the output of the `FacetOrderer` algorithm which is only available in the InstantSearch.

**Pull request merged**
